src/workers/postDownvoteCanceledRep.ts

```diff
@@ -0,0 +1,61 @@
+import {
+  ReputationEvent,
+  ReputationReason,
+  ReputationType,
+  Post,
+} from '../entity';
+import { messageToJson, Worker } from './worker';
+import { In } from 'typeorm';
+
+interface Data {
+  userId: string;
+  postId: string;
+}
+
+const worker: Worker = {
+  subscription: 'api.post-downvote-canceled-rep',
+  handler: async (message, con, logger): Promise<void> => {
+    const data: Data = messageToJson(message);
+    try {
+      const post = await con.getRepository(Post).findOneBy({ id: data.postId });
```

Comment:
Check that post exists, edge case but could happen with squad posts for example?

Reply:
I know you are checking below with `?.` but feels better to stop the workflow early if post can not be found.